experimental/form_app/lib/src/validation.dart

```diff
@@ -0,0 +1,161 @@
+import 'package:flutter/material.dart';
+import 'package:english_words/english_words.dart' as english_words;
+
+class FormValidationDemo extends StatefulWidget {
+  @override
+  _FormValidationDemoState createState() => _FormValidationDemoState();
+}
+
+class _FormValidationDemoState extends State<FormValidationDemo> {
+  final _formKey = GlobalKey<FormState>();
+  String adjective;
+  String noun;
+  bool agreedToTerms = false;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text('📖 Story Generator'),
+        actions: [
+          Padding(
+            padding: EdgeInsets.all(8),
+            child: FlatButton(
+              textColor: Colors.white,
+              child: Text('Submit'),
+              onPressed: () {
+                // Validate the form by getting the FormState from the GlobalKey
```

Comment:
Great place for a comment. I would have needed to check the docs to make sure I understood line 29.